test/let_value_test.cpp

```diff
@@ -264,6 +264,20 @@ TEST(Let, SimpleLetValueErrorWithAllocate) {
   })), std::invalid_argument);
 }
 
+TEST(Let, LetValueSuccessorWithException) {
+  EXPECT_THROW(sync_wait(unifex::just() | unifex::let_value([]() {
+    throw std::runtime_error("Throwing error for testing purposes");
+    return unifex::just();
+  })), std::runtime_error);
+}
+
+TEST(Let, LetValueWithException) {
```

Comment:
I think I'd leave this test out; it's semantically the same as the above test, and I think we have other tests that confirm the piping semantics.